Loop/Managers/LoopDataManager.swift

```diff
@@ -797,7 +793,7 @@ extension LoopDataManager {
         } else {
             storedCarbEntry = try await carbStore.addCarbEntry(carbEntry)
         }
-        self.temporaryPresetsManager.clearOverride(matching: .preMeal)
+        self.temporaryPresetsManager.clearOverride(matching: .preMeal, byEnteringMeal: true)
```

Comment:
Maybe make a separate method for this action; the 'byEnteringMeal' option is meaningless or confusing when passing in other override types.

Reply:
Can do!

Reply:
Done